test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -792,3 +792,129 @@ def main() -> None:
 
     with pytest.raises(errors.InconsistentNumQubits):
         main()
+
+
+def test_unnamed_retval_python_type() -> None:
+    """Tests subroutines which return unnamed Python values."""
+
+    @aq.function
+    def retval_test() -> int:
+        return 1
+
+    @aq.function
+    def caller() -> int:
+        return retval_test()
+
+    expected_qasm = """OPENQASM 3.0;
+def retval_test() -> int[32] {
+    int[32] retval_ = 1;
+    return retval_;
+}
+int[32] __int_1__ = 0;
+__int_1__ = retval_test();"""
+
+    assert caller().to_ir() == expected_qasm
+
+
+def test_unnamed_retval_qasm_type() -> None:
+    """Tests subroutines which return unnamed QASM values."""
+
+    @aq.function
+    def retval_test() -> aq.BitVar:
+        return aq.BitVar(1)
+
+    @aq.function
+    def caller() -> aq.BitVar:
+        return retval_test()
+
+    expected_qasm = """OPENQASM 3.0;
+def retval_test() -> bit {
+    bit retval_ = 1;
+    return retval_;
+}
+bit __bit_0__;
+__bit_0__ = retval_test();"""
+
+    assert caller().to_ir() == expected_qasm
+
+
+def test_recursive_unassigned_retval_python_type() -> None:
+    """Tests recursive subroutines which do not assign the return value to a variable."""
+
+    @aq.function
+    def retval_recursive() -> int:
+        retval_recursive()
```

Comment:
How to break your quantum computer 101 😄 hahaha

Reply:
yeah, this is proof that we don't actually execute these functions 😂